### DISCUSSION TITLE
Creating a dynamic JSON data based on User input

### DISCUSSION THREAD

**Discussion by Lokesh-Kanna-V** [vercel/next.js#62428] [Help]
### Summary

Hello all,
I am trying to build a dynamic UI for the project that I am working on. I am creating three login roles. One for admin, one for user and other for configurator. I am trying to store all the UI data in MongoDB so that I can map the contents of the UI data to create dynamic fields. 

Currently I am using useState to store the user input as below. But for the fixed fields, I have predefined the useState. If I am to dynamically create input fields (Creating an Input field when configurator Clicks on "Add Field" button, I will not be able to predefine the state variable. What can I do in this situation? 

How can I create a field dynamically and get the user input and store it dynamically (locally before pushing it to DB)?

**Comment by icyJoseph** [comment]
I guess, you can use a list to keep track of all the "added fields", and every time an new field is added you can do `setState(currentFields => [...currentFields, newField])`, where `newField` would be a data structure you define to keep track of the inputs on each field. 

Don't forget to make sure, that if newField is a const, it is not modified, for example by doing `{...newField}` in the previous operation, assuming level 1 depth on newField, you'd make sure to not mutate other inputs.
